tests/logrus_hook.go

```diff
@@ -74,10 +74,18 @@ var _ logrus.Hook = &logCache{}
 
 // attachLogCache sets logger to DebugLevel, attaches a LogCache hook and
 // returns it.
-func attachLogCache(logger *logrus.Logger) *logCache {
+func attachLogCache(fl logrus.FieldLogger) *logCache {
+	var logger *logrus.Logger
+	if ll, ok := fl.(*logrus.Logger); ok {
+		logger = ll
+	} else {
+		logger = logrus.New()
```

Comment:
Would it be a good idea to log a warning as we do in https://github.com/grafana/xk6-browser/pull/824 when `fl` is not the expected type?

Reply:
Tbh, I don't fully understand how it's used, so it's why i'm leaning towards making things clearer with a log message. But, if you know what's going on then i'm happy to leave it as is.

Reply:
Actually, reviewing this to better explain the context, I think the implementation is not exactly correct, as using the default logger here (in case the provided by `state.Logger` is not of type `*logrus.Logger`), will make the tests fail because that "default logger" is in the end not used by our implementation, so no hooks will be executed. This case is different from the one implemented in #824 where we actually replace one logger for another.

I think we should explicitly return an error if the given logger is not of type `*logrus.Logger` and fail the test, until we fix #818.
Let me make the changes for this :+1:

Reply:
Brilliant, i like it 🙂

Reply:
Pushed a new version in [d506083](https://github.com/grafana/xk6-browser/pull/842/commits/d506083ca71d697a3116c569133cfc603dd6c5ff).
Now, the tests will fail also if the given logger is not of type `*logrus.Logger`, but in this case the behavior is less misleading.
Currently the tests don't fail because we transform the "initial" `state.Logger` into a `*logrus.Logger` when our implementation of `log.New` is called, as modified in #824 .